components/org.wso2.carbon.identity.application.authenticator.oauth2/src/main/java/org.wso2.carbon.identity.application.authenticator.oauth2/Oauth2GenericAuthenticator.java

```diff
@@ -119,87 +162,82 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
                                                  AuthenticationContext context) throws AuthenticationFailedException {
 
         if (logger.isDebugEnabled()) {
-            logger.debug("processAuthenticationResponse");
+            logger.debug("Processing authentication response");
         }
-
         try {
-            Map<String, String> authenticatorProperties = context.getAuthenticatorProperties();
-            String clientId = authenticatorProperties.get(Oauth2GenericAuthenticatorConstants.CLIENT_ID);
-            String clientSecret = authenticatorProperties.get(Oauth2GenericAuthenticatorConstants.CLIENT_SECRET);
-            String redirectUri = authenticatorProperties.get(Oauth2GenericAuthenticatorConstants.CALLBACK_URL);
+            Map<String, String> authenticatorProperties = getAuthenticatorProperties(context);
+            String clientId = getClientId(authenticatorProperties);
+            String clientSecret = getClientSecret(authenticatorProperties);
+            String redirectUri = getCallbackURL(authenticatorProperties, request.getServerName(),
+                    request.getServerPort());
             Boolean basicAuthEnabled = Boolean.parseBoolean(
-                    authenticatorProperties.get(Oauth2GenericAuthenticatorConstants.IS_BASIC_AUTH_ENABLED));
+                    StringUtils.lowerCase(authenticatorProperties.get(IS_BASIC_AUTH_ENABLED)));
+            Boolean selfContainedTokenEnabled = Boolean.parseBoolean(
+                    StringUtils.lowerCase(authenticatorProperties.get(SELF_CONTAINED_TOKEN_ENABLED)));
             String code = getAuthorizationCode(request);
             String tokenEP = getTokenEndpoint(authenticatorProperties);
             String token = getToken(tokenEP, clientId, clientSecret, code, redirectUri, basicAuthEnabled);
-            String userInfoEP = getUserInfoEndpoint(authenticatorProperties);
-            String responseBody = getUserInfo(userInfoEP, token);
-
+            String userInfo = getUserInfo(selfContainedTokenEnabled, token, authenticatorProperties);
             if (logger.isDebugEnabled()) {
-                logger.debug("Get user info response : " + responseBody);
+                logger.debug("Get user info: " + userInfo);
             }
+            buildClaims(context, userInfo);
 
-            buildClaims(context, responseBody);
         } catch (ApplicationAuthenticatorException e) {
-            logger.error("Failed to process Connect response.", e);
+            throw new AuthenticationFailedException("Failed to process authentication response.", e);
+        } catch (MisconfigurationException e) {
+            logger.error(e.getMessage(), e);
             throw new AuthenticationFailedException(e.getMessage(), e);
         }
-
     }
 
     protected void buildClaims(AuthenticationContext context, String userInfoString)
-            throws ApplicationAuthenticatorException, AuthenticationFailedException {
```

Comment:
To preserve the backward compatibility lets not modify the method signature of public/protected methods

Reply:
This class has extended in two connectors(Naver and Kakao) and this change has affected to Naver connector. 
Since this change is addressing few issues we had with this generic connector, we decided to change that connector as well [PR for the Naver connector change](https://github.com/wso2-extensions/identity-outbound-auth-naver/pull/6 )